src/python-bindings/htcondor2/schedd.cpp

```diff
@@ -614,14 +614,26 @@ submitProcAds( int clusterID, long count, SubmitBlob * sb, ClassAd * & clusterAd
         }
 
         ClassAd * procAd = sb->make_job_ad( JOB_ID_KEY(clusterID, procID),
-            itemIndex, c, false, false, NULL, NULL );
+            itemIndex, c, false, spool, NULL, NULL );
         // FIXME: do something with sb->error_stack().
         if(! procAd) {
             // This was HTCondorInternalError in version 1.
             PyErr_SetString( PyExc_RuntimeError, "Failed to create job ad" );
             return -1;
         }
 
+        // If we're spooling, we'll need a copy of each proc ad later,
+        // so just store them now, rather than try to recreate them.
+        // Don't chain them to this clusterAd; the Python SubmitResult
+        // object will be giving us a copy of it when we need it.
+        if( spooledProcAds ) {
+            ClassAd * copy = new ClassAd( * procAd );
+            copy->Assign( ATTR_CLUSTER_ID, clusterID );
```

Comment:
proc ads should not have a cluster id attribute.  probably not a problem in practice though...

Reply:
For the record, I just I cargo-culted this in from https://github.com/htcondor/htcondor/blob/3997589a06f2bdda66a63260f49ecbf8e0b9853a/src/condor_submit.V6/submit.cpp#L2454 :)